tools/src/h5stat/h5stat.c

```diff
@@ -1639,6 +1647,19 @@ main(int argc, char *argv[])
             goto done;
         }
     }
+    else {
+        if ((fapl_id = h5tools_get_fapl(H5P_DEFAULT, NULL, NULL)) < 0) {
```

Comment:
Same comment here about this now always creating a fapl, which affects the logic used for `h5tools_fopen`. May need to set a boolean to signify that a custom driver was set so that can be used in the `h5tools_fopen` logic below rather than `(fapl_id != H5P_DEFAULT)`

Reply:
Agree that something needs changed.